### PR TITLE
Reorganize forms/pages fixes #1324

### LINKED ISSUE #1324: Reorganize Experiment form fields/pages
**Page 1: Overview**
- Type - Required - LABELED REQUIRED
- Name - Required - LABELED REQUIRED
- Short Description - Required - LABELED REQUIRED
- Data Science Bugzilla URL - Required - LABELED REQUIRED
- Experiment Owner - Required - LABELED REQUIRED
- Engineering Owner - optional
- Data Science Owner - Delayed required - LABELED REQUIRED
- Public Name - optional
- Public Description - optional
- Feature Bugzilla URL - optional
- Related Work URLs - optional
- Related Experiments? - optional

**Page 2: Timeline & Population** (New Page)
- Proposed Start Date - Delayed Required- LABELED REQUIRED
- Proposed Duration - Delayed Required- LABELED REQUIRED
- Proposed Enrollment - optional
- Population Size & Channel & Version - Delayed Required - LABELED REQUIRED
- Locales - optional
- Countries - optional
- Platform - optional
- Additional Filtering

Page 3: Design (Prefs/Addon/Branches) (same field behaviors)
- Same

Page 4: Objectives & Analysis (same field behaviors)
- Same

Page 5:  Risks & Testing (same field behaviors)
- Same

### PR DESCRIPTION
**Pull request by uhlissuh**
Fixes #1324 

This PR reorganizes many of the forms. Forms/pages affected are

-  overview (a reordering of fields and inclusion of data science owner on this page) 

- creation of a timeline/population page that pulls the timeline stuff from overview and the population stuff from variants/branches page

-  the branch/variants page getting paired down to just branch-y stuff.... 

- removal of data science owner from objectives/anaylsis form/page. 

<hr> 

I include screenshots of changes. The thing to look at here is the Timeline and Population page.... I used headlines and a separating line here to make two little sections.. this is a departure from some of the other styling, so I am fine with removing it and making one big hunk of form, I wouldn't say I hit the design out of the park necessarily... but I ended up splitting it just for visual clarity or something? Since these are kind of two distinct groups. Anyway, happy to change the look of the Timeline and Population page..

Screen Shot 2019-08-28 at 1 50 40 PM
<hr> 
Screen Shot 2019-08-28 at 1 54 02 PM
<hr> 
Screen Shot 2019-08-28 at 2 19 33 PM
<hr> 

There was a TON of test updating to do here... and adding tests for timeline/pop form and view.

### CODE REVIEW

File: app/experimenter/experiments/tests/test_forms.py [hunk 1]
```diff
@@ -679,143 +529,486 @@ def get_variants_form_data():
...
+        form = ExperimentTimelinePopulationForm(
+            request=self.request, data=self.data, instance=experiment
+        )
+        self.assertFalse(form.is_valid())
 
-        branch0 = experiment.variants.get(name=self.data["variants-0-name"])
-        self.assertTrue(branch0.is_control)
-        self.assertTrue(branch0.slug, "control-name")
-        self.assertEqual(branch0.ratio, 34)
+    def test_locales_choices(self):
```
[the remark this answers is not in the document]
Comment (jaredkerim) [reply app/experimenter/experiments/tests/test_forms.py]: Forget it forget this we can do this later.

File: app/experimenter/experiments/forms.py [hunk 2]
```diff
@@ -161,6 +161,17 @@ class ExperimentOverviewForm(
...
+        label="Short Description",
+        help_text=Experiment.SHORT_DESCRIPTION_HELP_TEXT,
+        widget=forms.Textarea(attrs={"rows": 3}),
+    )
+    data_science_bugzilla_url = BugzillaURLField(
+        label="Data Science Bugzilla URL",
+        help_text=Experiment.DATA_SCIENCE_BUGZILLA_HELP_TEXT,
+    )
     owner = forms.ModelChoiceField(
         required=True,
```
[the remark this answers is not in the document]
Comment (uhlissuh) [reply app/experimenter/experiments/forms.py]: it doesn't? data science bugzilla url in the screenshot does not say optional.

**Comment by jaredkerim** [comment]
@uhlissuh Swap `Engineering Owner` and `Data Science Owner` so all the required fields are together:

Screenshot 2019-09-04 15 11 20

File: tests/integration/pages/experiment_detail.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    _edit_branches_btn_locator = (By.CSS_SELECTOR,"#branches-edit-btn")
+
+    _page_wait_locator = (
+        By.CSS_SELECTOR,
+        "#overview",
+    )
+
+    def wait_for_page_to_load(self):
+        self.wait.until(
+            lambda _: "Overview"
```
Comment (jrbenny35) [inline tests/integration/pages/experiment_detail.py]: Looking at the errors on circleci [here](https://3247-71493337-gh.circle-artifacts.com/0/home/circleci/experimenter/tests/integration/integration-test.html) it looks like this doesn't work anymore. Overview was originally the first page in the create experiment flow. It appears the "Design" page is now loading. This should probably be changed to ```Design```

File: tests/integration/pages/experiment_detail.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+            in self.find_element(*self._page_wait_locator).text
+        )
+        return self
+
+    def click_edit(self):
+        self.find_element(*self._edit_branches_btn_locator).click()
+
+        from pages.experiment_branches import BranchesPage
+
+        return DetailPage(
```
Comment (jrbenny35) [inline tests/integration/pages/experiment_detail.py]: This function seems to just return this class. Looking at the test runs, the class needing to be returned should have a function named ```input_firefox_pref_name()```.

**Comment by jrbenny35** [comment]
You can also run the suite locally with ```make integration-test```, and after you do that once, you can just do ```tox -c tests/integration``` but you must have a python 3 env setup and tox installed.

File: tests/integration/pages/experiment_branches.py [hunk 5]
```diff
@@ -9,20 +9,14 @@
 from pages.base import Base
 
 
-class PopulationPage(Base):
+class BranchesPage(Base):
```
Comment (jaredkerim) [inline tests/integration/pages/experiment_branches.py]: Design Page

**Comment by uhlissuh** [comment]
Screen Shot 2019-09-09 at 4 19 10 PM 
I made the small changes we talked about... making sure teh order of fields is how we want it, and naming it design.

File: app/experimenter/templates/experiments/edit_nav_inline.html [hunk 6]
```diff
@@ -1,8 +1,8 @@
 <ul class="nav nav-pills nav-justified mb-3">
 
   <li class="nav-item">
-    <a 
-			class="nav-link {% if step == 1 %}active{% endif %}" 
+    <a
+			class="nav-link {% if step == 1 %}active{% endif %}"
```
Comment (jaredkerim) [inline app/experimenter/templates/experiments/edit_nav_inline.html]: Oops tab characters

File: tests/integration/pages/experiment_detail.py [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+from pages.base import Base
+
+class DetailPage(Base):
+
+    _edit_branches_btn_locator = (By.CSS_SELECTOR,"#branches-edit-btn")
+
+    _page_wait_locator = (
+        By.CSS_SELECTOR,
+        "body.page-detail-view",
```
Comment (jaredkerim) [inline tests/integration/pages/experiment_detail.py]: Beautiful 😻

**Pull request merged**